src/test/java/team/asd/service/ReservationServiceTest.java

```diff
@@ -1,50 +1,198 @@
 		assertThrows(ValidationException.class, () -> reservationService.readById(-11));
 	}
 
+	@Test
+	void testReadByIdNotNull() {
+		when(testReservationDao.readById(1)).thenReturn(TestResources.getTestResourcesReservation(1));
+		Reservation reservationMock = reservationService.readById(1);
+		Assertions.assertNotNull(reservationMock);
+		Assertions.assertEquals(reservationMock.getAgentId(), 54);
+		Assertions.assertEquals(reservationMock.getPrice(), 77.88);
+		verify(testReservationDao, atLeast(1)).readById(any(Integer.class));
+	}
+
 	@Test
 	void testCreateInCaseWhenOneOfParametersIsNull() {
+		assertThrows(ValidationException.class, () -> reservationService.create(null));
 		assertThrows(ValidationException.class, () -> reservationService.create(reservation));
 	}
 
+	@Test
+	void testCreateMock() {
+		doAnswer(invocation -> {
+			Reservation ReservationToCreate = invocation.getArgument(0, Reservation.class);
+			saveAndPopulateId(ReservationToCreate);
+			return null;
+		}).when(testReservationDao)
+				.create(any(Reservation.class));
+
+		assertNull(reservationMock);
+		reservationService.create(TestResources.getTestResourcesReservation(4));
+		assertNotNull(reservationMock);
+		assertNotNull(reservationMock.getId());
+		verify(testReservationDao, atLeast(1)).create(any(Reservation.class));
+	}
+
+	private void saveAndPopulateId(Reservation reservationToCreate) {
+		if (Objects.isNull(reservationToCreate)) {
+			return;
+		}
+		reservationToCreate.setId(1);
+		reservationMock = reservationToCreate;
+	}
+
 	@Test
 	void testUpdateInCaseWhenIdParameterIsNull() {
+		assertThrows(ValidationException.class, () -> reservationService.update(null));
 		assertThrows(ValidationException.class, () -> reservationService.update(reservation));
 	}
 
+	@Test
+	void testUpdateMock() {
+		reservationMock = TestResources.getTestResourcesReservation(2);
+
+		doAnswer(invocation -> Optional.ofNullable(invocation.getArgument(0, Reservation.class))
+				.map(this::updateReservation)
+				.orElse(null)).when(testReservationDao)
+				.update(any(Reservation.class));
+
+		reservationService.update(TestResources.getTestResourcesReservation(3));
+		Assertions.assertEquals(reservationMock.getPrice(), 66.5);
+		Assertions.assertEquals(reservationMock.getCustomerId(), 65);
+
+		Reservation reservationNew = TestResources.getTestResourcesReservation(2);
+		reservationNew.setPrice(8.88);
+		reservationNew.setCustomerId(99);
+		reservationService.update(reservationNew);
+		assertNotNull(reservationMock);
+		Assertions.assertEquals(reservationMock.getPrice(), 8.88);
+		Assertions.assertEquals(reservationMock.getCustomerId(), 99);
+
+		verify(testReservationDao, atLeast(2)).update(any(Reservation.class));
+	}
+
+	private Reservation updateReservation(Reservation reservation) {
+		if (Objects.isNull(reservation) || !reservationMock.getId()
+				.equals(reservation.getId())) {
+			return reservation;
+		}
+
+		if (!Objects.isNull(reservation.getOrganizationId())) {
```

Comment:
please use `Objects.nonNull` or even `someVariable != null` instead of `!Objects.isNull` to simplify the code